src/Item.cpp

```diff
@@ -4,26 +4,42 @@
 namespace tt
 {
 
-    Item::Item( sf::Texture texture,
+    Item::Item( std::string id,
+                sf::Texture texture,
                 const sf::Vector2i& size,
                 nl::json json)
         : AnimatedSprite(texture, size),
-            _json { json }
+            _json   { json },
+            _id     { id }
     {
 
     }
 
-    std::string Item::getDescription() { 
+    std::string Item::getID() { 
+        return _id;
+    }
+
+    std::string Item::getName() { 
 
         //
         // Why is it that the json object can be treated as a top level
         // associative array when this is created in ItemFactory but 
         // after it is assigned to the member variable in this object
         // instance, it becomes a json array at the top level, 
-        // and we must use _json[0] to get the
-        // first item?
+        // and we must use _json[0] to get the first item?
         //
+        return (std::string)(_json[0].at("name"));
+    }
+
+    std::string Item::getDescription() { 
         return (std::string)(_json[0].at("description"));
     }
 
+    bool Item::isObtainable() { 
+        if(_json[0]["obtainable"].is_null()) {
+            return false;
+        }
+        return (bool)(_json[0].at("obtainable"));
```

Comment:
Avoid C-style casts. Once I turn on strict compilation, this kind of thing will generate an error (really, it will generate a warning, but I will turn on "treat warnings as errors", too). Instead, C++ wants you to be explicit in the type of cast you want to do. In this case you could do:

```
return static_cast<bool>(_json[0].at("obtainable"));
```

Though, doesn't this JSON library allow you to do something like:

```
return _json[0].at("obtainable").as<bool>();
```

Or something like that...